File: sdk/dart/example/basic_example.dart
import 'dart:async';
import 'package:tungo_sdk/tungo.dart';

void main() async {
  // Create client with options
  final options = TunGoOptions(
    localPort: 3000,
    subdomain: 'my-dart-app',
    logLevel: 'INFO',
  );

  // Set up event listeners
  final eventListener = TunGoEventListener(
    onConnect: (info) {
      print('✓ Tunnel established!');
      print('  Public URL: ${info.url}');
      print('  Subdomain: ${info.subdomain}');
      print('  Client ID: ${info.clientId}');
      print('\nTunnel is running. Press Ctrl+C to stop.');
      print('Try accessing: ${info.url}');
    },
    onDisconnect: (reason) {
      print('✗ Disconnected: $reason');
    },
    onError: (error) {
      print('✗ Error: $error');
    },
    onReconnect: (attempt, maxRetries) {
      print('↻ Reconnecting (attempt $attempt/$maxRetries)...');
    },
  );

  final client = TunGoClient(options, eventListener);

  // Start tunnel
  try {
    await client.start();

    // Keep running
    await Completer<void>().future;
  } catch (e) {
    print('Failed to start tunnel: $e');
  }
}

File: sdk/java/src/main/java/com/cubetiqs/tungo/examples/BasicExample.java
package com.cubetiqs.tungo.examples;

import com.cubetiqs.tungo.TunGoClient;
import com.cubetiqs.tungo.TunGoEventListener;
import com.cubetiqs.tungo.TunGoOptions;
import com.cubetiqs.tungo.TunnelInfo;

/**
 * Basic example of using TunGo SDK
 */
public class BasicExample {
    public static void main(String[] args) {
        // Configure options
        TunGoOptions options = TunGoOptions.builder(3000)
            .serverHost("localhost")
            .controlPort(5555)
            .subdomain("my-custom-subdomain")
            .logLevel("INFO")
            .build();

        // Create event listener
        TunGoEventListener listener = new TunGoEventListener() {
            @Override
            public void onConnect(TunnelInfo tunnelInfo) {
                System.out.println("✓ Tunnel established!");
                System.out.println("  Public URL: " + tunnelInfo.getUrl());
                System.out.println("  Subdomain: " + tunnelInfo.getSubdomain());
                System.out.println("  Client ID: " + tunnelInfo.getClientId());
            }

            @Override
            public void onDisconnect(String reason) {
                System.out.println("✗ Disconnected: " + reason);
            }

            @Override
            public void onError(Throwable error) {
                System.err.println("✗ Error: " + error.getMessage());
            }

            @Override
            public void onReconnect(int attempt, int maxRetries) {
                System.out.println("↻ Reconnecting (attempt " + attempt + "/" + maxRetries + ")...");
            }

            @Override
            public void onStatus(String message) {
                System.out.println("ℹ Status: " + message);
            }
        };

        // Create client
        TunGoClient client = new TunGoClient(options, listener);

        // Add shutdown hook
        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            System.out.println("\nShutting down...");
            client.shutdown();
        }));

        try {
            // Start tunnel
            System.out.println("Starting TunGo tunnel...");
            TunnelInfo info = client.start();
            
            System.out.println("\nTunnel is running. Press Ctrl+C to stop.");
            System.out.println("Try accessing: " + info.getUrl());

            // Keep running
            Thread.currentThread().join();

        } catch (Exception e) {
            System.err.println("Failed to start tunnel: " + e.getMessage());
            e.printStackTrace();
            System.exit(1);
        }
    }
}
